packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart

```diff
@@ -0,0 +1,151 @@
+/// web-browsing experience, and presents HTML, CSS, and JavaScript content
+/// alongside your app’s native views. Use it when web technologies satisfy your
+/// app’s layout and styling requirements more readily than native views. For
+/// example, you might use it when your app’s content changes frequently.
+///
+/// A web view offers control over the navigation and user experience through
+/// delegate objects. Use the navigation delegate to react when the user clicks
+/// links in your web content, or interacts with the content in a way that
+/// affects navigation. For example, you might prevent the user from navigating
+/// to new content unless specific conditions are met. Use the UI delegate to
+/// present native UI elements, such as alerts or contextual menus, in response
+/// to interactions with your web content.
+///
+/// For more extensive customizations, create your web view using a
+/// [WebViewConfiguration] object. For example, use a web view configuration
+/// object to specify handlers for custom URL schemes, manage cookies, and
+/// customize preferences for your web content.
+///
+/// Before your web view appears onscreen, load content from a web server using
+/// a [UrlRequest] class or load content directly from a local file or HTML
+/// string. The web view automatically loads embedded resources such as images
+/// or videos as part of the initial load request. It then renders your content
+/// and displays the results inside the view’s bounds rectangle.
+///
+/// A web view automatically converts telephone numbers that appear in web
+/// content to Phone links. When the user taps a Phone link, the Phone app
+/// launches and dials the number. Use the [WebViewConfiguration] object to
+/// change the default data detector behavior.
+///
+/// **Managing the Navigation Through Your Web Content**
+/// [WebView[ provides a complete browsing experience, including the ability to
+/// navigate between different webpages using links, forward and back buttons,
+/// and more. When the user clicks a link in your content, the web view acts
+/// like a browser and displays the content at that link. To disallow
+/// navigation, or to customize your web view’s navigation behavior,
+/// provide your web view with a navigation delegate — that is, an object that
+/// extends the [NavigationDelegate] class. Use your navigation delegate to
+/// modify the web view’s navigation behavior, or to track the loading progress
+/// of new content.
+///
+/// You can also use the methods of [WebView] to navigate programmatically
+/// through your content, or to trigger navigation from other parts of your
+/// app’s interface. For example, if your UI includes forward and back buttons,
+/// connect those buttons to the [goBack] and [goForward] methods of your web
+/// view to trigger the corresponding web navigation. Use the [canGoBack] and
+/// [canGoForward] fields to determine when to enable or disable your buttons.
```

Comment:
We should not be duplicating docs wholesale from Apple code.

I would much prefer we say something like:
"Wraps WKWebView. See insert-link-here for documentation".

Reply:
I was back and forth about to what extent we should document these classes. I think this is a good pattern moving forward. I think this would also pair well with testing that links are up to date. I created an issue for it: https://github.com/flutter/flutter/issues/96974https://github.com/flutter/flutter/issues/96974